pkg/cloud/services/eks/cluster.go

```diff
@@ -707,12 +707,19 @@ func compareEncryptionConfig(updatedEncryptionConfig, existingEncryptionConfig [
 		return false
 	}
 	for index, encryptionConfig := range updatedEncryptionConfig {
-		if encryptionConfig.Provider != existingEncryptionConfig[index].Provider {
+		if getKeyArn(encryptionConfig) != getKeyArn(existingEncryptionConfig[index]) {
 			return false
 		}
-		if cmp.Equals(encryptionConfig.Resources, existingEncryptionConfig[index].Resources) {
+		if !cmp.Equals(encryptionConfig.Resources, existingEncryptionConfig[index].Resources) {
 			return false
 		}
 	}
 	return true
 }
+
+func getKeyArn(encryptionConfig *eks.EncryptionConfig) string {
+	if encryptionConfig.Provider != nil && encryptionConfig.Provider.KeyArn != nil {
+		return *encryptionConfig.Provider.KeyArn
```

Comment:
`aws.StringValue` deals with nil case as well. Since KeyArn sdk field, better to use SDK dereferencing.

Reply:
Ah nice, I didn't know that handled nils. Will update

Reply:
Updated. I left the nil check on `Provider` in there as I believe that could still be a nil pointer.